peers/app_request_network.go

```diff
@@ -59,7 +56,8 @@ func NewNetwork(
 		),
 	)
 
-	networkID, err := infoClient.GetNetworkID(context.Background())
+	infoClient := cfg.InfoAPI.GetClient()
```

Comment:
Calls to `info.Peers`, `info.GetNodeID`, and `info.GetNodeIP` are not using the configured `rpc.Option`s. 

We should compose `AppRequestNetwork` with `InfoAPI` rather than `info.Client` to make it more obvious that `InfoApi.GetClient` should be used in conjunction with `InfoApi.GetOptions`.